core/src/workflows/hold_entry_remove.rs

```diff
@@ -0,0 +1,48 @@
+use crate::{
+    context::Context,
+    dht::actions::remove_entry::remove_entry,
+    network::{
+        actions::get_validation_package::get_validation_package, entry_with_header::EntryWithHeader,
+    },
+    nucleus::validation::validate_entry,
+};
+
+use holochain_core_types::{
+    cas::content::AddressableContent,
+    entry::Entry,
+    error::HolochainError,
+    validation::{EntryAction, EntryLifecycle, ValidationData},
+};
+use std::sync::Arc;
+
+pub async fn hold_remove_workflow<'a>(
+    entry_with_header: EntryWithHeader,
+    context: Arc<Context>,
+) -> Result<(), HolochainError> {
+    let EntryWithHeader { entry, header } = &entry_with_header;
+
+    // 1. Get validation package from source
+    let maybe_validation_package = await!(get_validation_package(header.clone(), &context))?;
+    let validation_package = maybe_validation_package
+        .ok_or("Could not get validation package from source".to_string())?;
+
+    // 2. Create validation data struct
+    let validation_data = ValidationData {
+        package: validation_package,
+        lifecycle: EntryLifecycle::Dht,
```

Comment:
This has to be meta:
```suggestion
        lifecycle: EntryLifecycle::Meta,
```

...which reminds me that we do need set the action right when the lifecycle actually is `Dht`, which is the case for storing the deletion entry, which happens in the regular `hold_entry_workflow`. I guess we need to test if the entry is a deletion entry, or if it is an update entry, and set the action there accordingly.